### ISSUE TITLE
Quarkus DEV mode requires 3.9.6 Maven

### ISSUE THREAD

**Issue by michalvavrik** [quarkusio/quarkus#42478]
### Describe the bug

After https://github.com/quarkusio/quarkus/pull/41648 Quarkus now requires Maven 3.9.6 in DEV mode. I don't see this documented in migration guide and from what I understand from PR title, it is Quarkus that requires 3.9.6, not the Quarkus application.

### Expected behavior

Fix requirement or document the situation.

### Actual behavior

Exception is raised: 

```
[mvavrik@apc-ap8641-3a-d11d code-with-quarkus]$ mvn quarkus:dev
[INFO] Scanning for projects...
[INFO] 
[INFO] ---------------------< org.acme:code-with-quarkus >---------------------
[INFO] Building code-with-quarkus 1.0.0-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- quarkus-maven-plugin:999-SNAPSHOT:dev (default-cli) @ code-with-quarkus ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.874 s
[INFO] Finished at: 2024-08-12T11:33:09+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:dev (default-cli) on project code-with-quarkus: Detected Maven Version (3.8.8)  is not supported, it must be in [3.9.6,). -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

### How to Reproduce?

Steps to reproduce the problem:

```
quarkus create app
cd code-with-quarkus/
sed -i -e "s|<quarkus.platform.group-id>io.quarkus.platform</quarkus.platform.group-id>|<quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>|" pom.xml
sed -i -e "s|<quarkus.platform.version>3.13.2</quarkus.platform.version>|<quarkus.platform.version>999-SNAPSHOT</quarkus.platform.version>|" pom.xml
mvn -DskipTests -DskipITs clean install
mvn quarkus:dev
```

### Output of `uname -a` or `ver`

Fedora 40

### Output of `java -version`

OpenJDK Runtime Environment Temurin-21.0.1+12

### Quarkus version or git rev

Quarkus HEAD 7d88a939c70 (999-SNAPSHOT)

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.8 (4c87b05d9aedce574290d1acc98575ed5eb6cd39)

### Additional information

Quarkus CLI 3.12.2.

**Comment by michalvavrik** [comment]
\cc @gsmet @manofthepeace

**Comment by gsmet** [comment]
@michalvavrik could you post the full stacktrace with `-e`? I wonder what's triggering this.

**Comment by gsmet** [comment]
Ah wait, you have been kind enough to provide a scripted reproducer, let me try.

```
Caused by: org.apache.maven.plugin.MojoExecutionException: Detected Maven Version (3.8.7)  is not supported, it must be in [3.9.6,).
    at io.quarkus.maven.components.MavenVersionEnforcer.enforce (MavenVersionEnforcer.java:71)
    at io.quarkus.maven.components.MavenVersionEnforcer.ensureMavenVersion (MavenVersionEnforcer.java:31)
    at io.quarkus.maven.DevMojo.execute (DevMojo.java:408)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:298)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

**Comment by michalvavrik** [comment]
> @michalvavrik thanks for your vigilance, this was an unintended side effect. I fixed it here: #42481 .

got it, thanks
